gtsfm/utils/io.py

```diff
@@ -213,19 +265,39 @@ def write_points(gtsfm_data: GtsfmData, images: List[Image], save_dir: str) -> N
         # TODO: assign unique indices to all keypoints (2d points)
         point2d_idx = 0
 
+        # Iterate through track to gather a list of 3D points forming the point cloud.
+        point_cloud_list = []
+        for j in range(num_pts):
+            track = gtsfm_data.get_track(j)
+            x, y, z = track.point3()
+            point_cloud_list.append([x, y, z])
+
+        # Transform the point cloud to be aligned with x,y,z axes using SVD.
+        point_cloud = np.array([np.array(p) for p in point_cloud_list])  # point_cloud has shape Nx3
+        points_centered, means = ellipsoid_utils.center_point_cloud(point_cloud)
+        points_filtered = ellipsoid_utils.filter_outlier_points(points_centered)
+        rot = ellipsoid_utils.get_rotation_matrix(points_filtered)
+        aligned_points = ellipsoid_utils.apply_ellipsoid_rotation(rot, points_centered)
+
```

Comment:
maybe have the transform function which returns a `GtsfmData`, and the write function just write the object.

Reply:
sounds good. I'll make some sort of wrapper function in `ellipsoid.py` which contains all this transformation code. That way, `write_points()` within `io.py` can just call the wrapper function and write the object.